### PR TITLE
Create Load_VIP_DeathInformantPr.sql

### PR DESCRIPTION
**Pull request by frahmanr**
10/14/2022 First version

### CODE REVIEW

File: RVRS_Staging/Database/Procedure/Load_VIP_DeathInformantPr.sql [hunk 1]
```diff
@@ -0,0 +1,806 @@
...
+PRINT '6'  + CONVERT (VARCHAR(50),GETDATE(),109)	
+					
+   
+		SELECT *
+			
+	        ,CASE WHEN LEFT(FirstName,1) LIKE '[^a-zA-Z]' AND FirstName NOT LIKE '-%' THEN 'FirstName|Error:Not a valid First Name' ELSE '' END AS LoadNote_1
+	        ,CASE WHEN LEFT(MiddleName,1) LIKE '[^a-zA-Z]' AND MiddleName NOT LIKE '-%' THEN 'MiddleName|Error:Not a valid Middle Name' ELSE '' END AS LoadNote_2
+	        ,CASE WHEN LEFT(LastName,1) LIKE '[^a-zA-Z]' AND LastName NOT LIKE '-%' THEN 'LastName|Error:Not a valid Last Name' ELSE '' END AS LoadNote_3
+	        ,CASE WHEN OtherInformantRelation != 'NULL' AND InformantRelation !='OTHER' THEN 'InformantRelation,OtherInformantRelation|Error:When Other is not blank relation should be Other' ELSE '' END AS LoadNote_4
+
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/Load_VIP_DeathInformantPr.sql:226]: Add one validation that compare code in INFO_RELATION_LIST and INFO_RELATION

File: RVRS_Staging/Database/Procedure/Load_VIP_DeathInformantPr.sql [hunk 2]
```diff
@@ -0,0 +1,806 @@
...
+	
+SET @LastLoadedDate=(SELECT MAX(LastLoadDate) FROM [RVRS_testdb].[RVRS].[Execution] WITH(NOLOCK) WHERE Entity='DeathInformant' AND ExecutionStatus='Completed')
+	        IF @LastLoadedDate IS NULL SET @LastLoadedDate = '01/01/1900'
+PRINT '2'  + CONVERT (VARCHAR(50),GETDATE(),109)
+	
+	
+			
+
+		        SELECT TOP 10000000  D.DEATH_REC_ID AS SrId
+					  ,P.PersonId ,INFO_NME FirstName,INFO_LST_NME LastName,INFO_MIDD_NME MiddleName,ISNULL(INFO_SUFFIX,'NULL') Suffix,INFO_RELATION_LIST RelationList,ISNULL(INFO_RELATION,'NULL') InformantRelation,ISNULL(INFO_RELATION_OTHER,'NULL') OtherInformantRelation
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/Load_VIP_DeathInformantPr.sql:134]: Replace iSNULL by COALESCE

**Review by almabekwoldesenbet — CHANGES_REQUESTED** [review]
Modified Code layout accordingly 

To generate the prod version 
  Set three variables
-    DECLARE     @Workdb VARCHAR(128) = '[RVRS_testdb].',  -- For PROD leave this blank or set to @stagingdb 
-   DECLARE    @targetloc varchar(128) =@Workdb --'[RVRS_PROD].[RVRS_ODS].' For PROD replace this by [RVRS_PROD].[RVRS_ODS]. 
- @testflag bit = 1 -- set this to 0 for production 

**then** 
Load data to ODS  

**Send final code review**

File: RVRS_Staging/Database/Procedure/Load_VIP_DeathInformantPr.sql (hunk 1 above)
Comment (frahmanr) [reply RVRS_Staging/Database/Procedure/Load_VIP_DeathInformantPr.sql:232]: Fixed it.

File: RVRS_Staging/Database/Procedure/Load_VIP_DeathInformantPr.sql (hunk 2 above)
Comment (frahmanr) [reply RVRS_Staging/Database/Procedure/Load_VIP_DeathInformantPr.sql]: Fixed it.

File: RVRS_Staging/Database/Procedure/Load_VIP_DeathInformantPr.sql [hunk 3]
```diff
@@ -123,15 +123,15 @@ INSERT INTO [RVRS_testdb].[RVRS].[Execution]
...
-SET @LastLoadedDate=(SELECT MAX(LastLoadDate) FROM [RVRS_testdb].[RVRS].[Execution] WITH(NOLOCK) WHERE Entity='DeathInformant' AND ExecutionStatus='Completed')
+SET @LastLoadedDate=(SELECT MAX(LastLoadDate) FROM [RVRS].[Execution] WITH(NOLOCK) WHERE Entity='DeathInformant' AND ExecutionStatus='Completed')
 	        IF @LastLoadedDate IS NULL SET @LastLoadedDate = '01/01/1900'
 PRINT '2'  + CONVERT (VARCHAR(50),GETDATE(),109)
 	
 	
 			
 
 		        SELECT TOP 10000000  D.DEATH_REC_ID AS SrId
-					  ,P.PersonId ,INFO_NME FirstName,INFO_LST_NME LastName,INFO_MIDD_NME MiddleName,ISNULL(INFO_SUFFIX,'NULL') Suffix,INFO_RELATION_LIST RelationList,ISNULL(INFO_RELATION,'NULL') InformantRelation,ISNULL(INFO_RELATION_OTHER,'NULL') OtherInformantRelation
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/Load_VIP_DeathInformantPr.sql]: Remove TOP 10000000
@top VARCHAR(128) =  'TOP 10000000 ' -- For PROD leave this blank

File: RVRS_Staging/Database/Procedure/Load_VIP_DeathInformantPr.sql (hunk 1 above)
Comment (almabekwoldesenbet) [reply RVRS_Staging/Database/Procedure/Load_VIP_DeathInformantPr.sql:232]: Add Bracket (InformantRelation IS NULL AND RelationList  IS NOT NULL) THEN

OR do this 

Add one more variable on code layout : @AdditionalJoin  VARCHAR (MAX) = 

'LEFT JOIN [RVRS_PROD].[RVRS_ODS].[RVRS].[DimInformantRelation] DS on convert (int,d.INFO_RELATION_LIST) = ds.Code and 

And add below replace all the 17 OR entries by 
COALESCE (d.InformantRelation, '')  <> COALESCE (ds.InformantRelationDesc , '')

@AdditionalJoin  should be added at SELECT statement Line between 388 and 389

**Review by almabekwoldesenbet — CHANGES_REQUESTED** [review]
Data change 
MiddleName 

-  Add   - conversion table 
Informant Other 
- remove Duplicate 
OtherInformantRelationDesc	(No column name)
COUSIN	2
DAUGHTER-IN-LAW	2
EX-WIFE	2
EX-WIFE,HCP	2
GRANDDAUGHTER-IN-LAW	2
MATERNAL GRANDMOTHER	2
PERSONAL REPRESENTATIVE	2
PERSONAL RESPRESENTATIVE	3
SIGNIFICANT OTHER	2
STEP-CHILD	2
STEP-DAUGHTER POA	2
STEP-FATHER/LEGAL GUARDIAN	2
STEP-SON	2

**Comment by almabekwoldesenbet** [comment]
10-21-2022 - RVRS_169 was completed. First load to ODS about 466699 records were loaded successfully

**Pull request merged**